src/community/label_propagation.c

```diff
@@ -274,6 +276,13 @@ int igraph_community_label_propagation(const igraph_t *graph,
         RNG_END();
     }
 
+    if (weights) {
+        igraph_inclist_destroy(&il);
+    } else {
+        igraph_adjlist_destroy(&al);
+    }
+    IGRAPH_FINALLY_CLEAN(1);
+
     /* Shift back the membership vector, permute labels in increasing order */
     /* We recycle label_counters here :) */
     igraph_vector_fill(&label_counters, -1);
```

Comment:
`unlabelled_left` is initialized up at the top of the function but it is not used up to this point. I'd leave the _declaration_ up at the top and move the _initialization_ here so it stays closer to the for loop where its "real" value is being computed. (In other words, the `for` loop has a pre-condition that `unlabelled_left` is false, and it is easier to see the pre-condition if it is declared here and not at the top).